content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-bitbucket-identity.md

```diff
@@ -0,0 +1,422 @@
+  }
+}
+
+output "bitbucket-identity" {
+  value = chainguard_identity.bitbucket.id
+}
+
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.bitbucket.id
+  group    = chainguard_group.user-group.id
+  role     = data.chainguard_roles.viewer.items[0].id
+}
+EOF
+```
+
+Following that, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+
+## Creating Your Resources
+
+First, run `terraform init` to initialize Terraform's working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you're satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 4 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + bitbucket-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `bitbucket-identity` value. Note that you may receive a `PermissionDenied` error part way through the apply step. If so, run `chainctl auth login` once more, and then `terraform apply` again to resume creating the identity and resources.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+bitbucket-identity = "<your bitbucket identity>"
+```
+
+This is the identity's [UIDP (unique identity path)](/chainguard/chainguard-enforce/reference/events/#uidp-identifiers), which you configured the `bitbucket.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Bitbucket workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
+
+```sh
+chainctl iam identities ls
+```
+
+You're now ready to edit a Bitbucket pipeline in order to test out this identity.
+
+
+## Testing the identity with a Bitbucket pipeline
+
+To test the identity you created with Terraform in the previous section, ensure you have Pipelines enabled for your repository and then create a `bitbucket-pipelines.yml` file in the root of your repository. Note: if you already have a pipeline with steps defined then you only need to add the `oidc: true` field to your pipeline to enable OIDC for the step in question.
```

Comment:
I might even specify in the prereqs that folks ought to enable pipelines. This had me scratching my head for a bit

Reply:
Good call out, I added a link to the guide on enabling them.